src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -1027,7 +1027,7 @@ public static int IndexOf<T>(T[] array, T value, int startIndex, int count)
             if (typeof(T) == typeof(byte))
```

Comment:
Are the IndexOf/LastIndexOf methods close to inlining budget?
Would it help to move the code paths after this line to a different method, annotate that one with AggressiveInlining?
Last time I checked, the JIT does not really know that the other if code path is not possible during inlining and with that overestimates the size of this method.

Reply:
Likely over, too many argument checks and throw calls (3 if blocks, 3 method calls before it gets to the IndexOf call). To bring it down would want to combine all the argument checks into a single if block; then have the single throwhelper call determine which one to throw.

Though probably not worth it.